### PR TITLE
GitPod integration setup

### LINKED ISSUE #546: [NV-388] - GitPod integration setup
## Why?

GitPod can new contributors setup the project faster and to avoid versioning issues for global dependencies such as Node, pnpm and etc…

## What?

Currently, we have partial support, but looks like a couple of things are missing:

* When compiling the project during rebuild time, we need to run the `npm run setup:project` command instead of just running npm install in the root
* When running the service URL outside of the gitpod tab, services are trying to access the local service using localhost:3000 which is not working since localhost only exists when running inside the vscode browser.  (We need to create a new .env file configuration for GitPod and somehow inject the URLs of the services)
* Consider creating a bootstrap script that creates a default user inside the web platform the GitPod can automatically log in to.
* Currently, yarn start running all projects including the CLI. We should only run: API, WEB, WIDGET, WS and their dependencies
* Any other things we should consider for a better experience?

Current GitPod config runs here: [https://github.com/novuhq/novu/blob/main/.gitpod.yml](https://github.com/novuhq/novu/blob/main/.gitpod.yml)

### PR DESCRIPTION
**Pull request by vrabe**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix
- **Why was this change needed?** (You can also link to an open issue here)
#546
- **Other information**:
Bootstrap script is not done yet.

### CODE REVIEW

File: .gitpod.yml [hunk 1]
```diff
@@ -1,28 +1,54 @@
...
+      pnpm envsetup && pnpm start
+  - name: WIDGET
+    init: gp sync-await setup
+    command: |
+      cd apps/widget
+      API_URL=`gp url 3000`
+      WS_URL=`gp url 3002`
+      sed -i "/REACT_APP_API_URL=/c\REACT_APP_API_URL=$API_URL" .env
+      sed -i "/REACT_APP_WS_URL=/c\REACT_APP_WS_URL=$WS_URL" .env
+      pnpm envsetup && pnpm start:dev
```
Comment (p-fernandez) [inline .gitpod.yml:38]:
```suggestion
      pnpm envsetup && pnpm start
```

File: .gitpod.yml [hunk 2]
```diff
@@ -1,28 +1,54 @@
...
 ports:
   - port: 4200
     onOpen: open-preview
+    visibility: public
```
Comment (p-fernandez) [inline .gitpod.yml:42]: What's the reason of making the port public? Wouldn't this make it accessible to outside of GitPod to anyone with the URL?

Comment (vrabe) [reply .gitpod.yml:42]: This port can be private. But if WS and API ports are private, we will need to deal with CORS problem.

File: .gitpod.yml (hunk 1 above)
Comment (vrabe) [reply .gitpod.yml:38]: It's default port is 3000, which is used by API. If we don't set to other port, there will be collision.

Comment (p-fernandez) [reply .gitpod.yml:38]: Then it feels right to update the Widget in the script also to provide a different port.

Comment (vrabe) [reply .gitpod.yml:38]: like this?
```suggestion
      PORT=4500
      pnpm envsetup && pnpm start
```

File: .gitpod.yml [hunk 3]
```diff
@@ -1,28 +1,54 @@
...
     command: cd apps/api && pnpm start
   - name: WS
+    init: gp sync-await setup
     command: cd apps/ws && pnpm start
   - name: WEB
-    command: cd apps/web && pnpm start
+    init: gp sync-await setup
+    command: |
+      cd apps/web
+      API_URL=`gp url 3000`
```
Comment (scopsy) [inline .gitpod.yml:27]: I think the WS_URL is missing on the web since the client tries to access localhost when visiting the web platform.

**Comment by vrabe** [comment]
It's better to make this pull request merged first and make another pull request for the bootstrap script later.

**Pull request merged**